kyuubi-common/src/main/scala/org/apache/kyuubi/util/RowSetUtils.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.util
+
+import java.nio.ByteBuffer
+import java.time.chrono.IsoChronology
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+import java.time.temporal.ChronoField
+import java.util.Locale
+
+import scala.language.implicitConversions
+
+object RowSetUtils {
+
+  lazy val dateFormatter = {
+    createDateTimeFormatterBuilder()
+      .appendPattern("yyyy-MM-dd")
+      .toFormatter(Locale.US)
+      .withChronology(IsoChronology.INSTANCE)
+  }
+
+  lazy val timeFormatter: DateTimeFormatter = {
+    createDateTimeFormatterBuilder()
+      .appendPattern("HH:mm:ss")
+      .toFormatter(Locale.US)
+      .withChronology(IsoChronology.INSTANCE)
+  }
+
+  lazy val timestampFormatter: DateTimeFormatter = {
+    createDateTimeFormatterBuilder()
+      .appendPattern("yyyy-MM-dd HH:mm:ss")
+      .appendFraction(ChronoField.NANO_OF_SECOND, 0, 9, true)
+      .toFormatter(Locale.US)
+      .withChronology(IsoChronology.INSTANCE)
+  }
+
+  lazy val timestampWithZoneFormatter: DateTimeFormatter = {
+    createDateTimeFormatterBuilder()
+      .appendPattern("yyyy-MM-dd HH:mm:ss")
+      .appendFraction(ChronoField.NANO_OF_SECOND, 0, 9, true)
+      .optionalStart()
+      .appendOffsetId()
+      .appendLiteral('[')
+      .parseCaseSensitive()
+      .appendZoneRegionId()
+      .appendLiteral(']')
```

Comment:
this part is optional, as a datetime formatter, it will never print this zone part here, right?

Reply:
> this part is optional, as a datetime formatter, it will never print this zone part here, right?

It will print, the result is `2018-11-17 13:33:33.6+08:00[Asia/Shanghai]`. The optional here means that if append zone part failed, the result will print without it. Just for fault-tolerant.

Reply:
do we have a test for zone filed check？

Reply:
It seems that the fail can not happen here, will remove `optionalStart` part.

Reply:
`2018-11-17 13:33:33.6+08:00[Asia/Shanghai]` looks weird, we add the offset-based and region-based TZs in the same timestamp, we shall keep only one